### DISCUSSION TITLE
Cannot view chart in Jetpack Compose Preview

### DISCUSSION THREAD

**Discussion by anchithAcharya** [patrykandpatrick/vico#805] [Ideas]
### How to reproduce

Run the following code in an android project with Jetpack Compose set up:
``` kotlin
@Preview
@Composable
fun ChartComposable() {
    val chart = rememberCartesianChart(
        rememberLineCartesianLayer(),
        startAxis = rememberStartAxis(),
        bottomAxis = rememberBottomAxis(),
    )
    val cartesianChartModelProducer = remember { CartesianChartModelProducer() }
    LaunchedEffect(Unit) {
        cartesianChartModelProducer.runTransaction {
            lineSeries {
                series(listOf(1, 2, 3, 4))
            }
        }
    }
    CartesianChartHost(
        chart = chart,
        modelProducer = cartesianChartModelProducer
    )
}
```

### Observed behavior

Preview fails to load, giving the following exception:
`java.lang.ClassNotFoundException: com.patrykandpatrick.vico.core.cartesian.data.CartesianChartModelProducer`

### Expected behavior

Preview should be visible

### Vico version(s)

2.0.0-alpha.23

### Android version(s)

SDK version 34, minSDK 30

**Comment by Gowsky** [comment]
Hi @anchithAcharya, we ran your code and we didn't experience the exception. It seems to be a build cache issue. It should disappear when you rebuild the project. However, the preview will remain blank, as Compose previews don't support asynchronous execution (`CartesianChartModelProducer` runs off the main thread). We'll check if we can move the execution to the main thread for Compose previews. In the meantime, you can create the [model synchronously](https://patrykandpatrick.com/vico/wiki/cartesian-charts/data#manual-cartesianchartmodel-creation) **for the preview**.

Since this is not entirely a bug, I'm converting this to a discussion.

**Reply by anchithAcharya** [reply]
Hi @Gowsky, it was indeed a build cache issue! Thank you, and apologies 😅. The discussion you linked to seems to not exist though.

**Reply by Gowsky** [reply]
@anchithAcharya, apologies. I fixed the link.

**Reply by cristianboicu01** [reply]
Hello! I've run into the same issue. Compose preview does not render the chart. Tried to invalidate cache and restart but that didn't help. I'm not sure how to create the model synchronously and the link is broken again. Can you update the link or provide some information on this topic, please?

P.S. I've managed to render the chart in preview by changing the Dispatchers to Main in CartesianChartModelProducer but I don't know if this is achievable through the API.

**Reply by Gowsky** [reply]
Hello, @cristianboicu01. Please see sections 2.2.5, 2.4.3, 2.5.3, and 2.6.3 of [the guide](https://www.patrykandpatrick.com/vico/guide/latest) and the `model` parameter of `CartesianChartHost`.

**Reply by Shusek** [reply]
@Gowsky Would it be possible to provide a parameter for setting the Dispatcher? It would help in cases where we already have a complete compose element, and we just want to preview it without rewriting the entire logic from scratch.

**Comment by Gowsky** [comment]
[Vico 2.1.0-alpha-3](https://github.com/patrykandpatrick/vico/releases/tag/v2.1.0-alpha.3) brings the support for previews using `CartesianChartHost` composable and `CartesianChartModelProducer`. Please see the release notes for details.

**Reply by ghost** [reply]
Thanks for the update, I am rewriting 1.14.0 to v2 and now the preview works fine.

However, compose screenshot testing is still not working (it was working fine with **vico 1.14.0**).
I took a preview from `com.patrykandpatrick.vico.sample.showcase.charts#BasicColumnChartPreview` and put it in test class and nothing was generated.
Instruction: https://developer.android.com/studio/preview/compose-screenshot-testing

Is there a chance that you can take a look at this?

**Reply by Gowsky** [reply]
Hello @sylwia-golonko. Thank you for bringing this into our attention. This will be addressed in the next release. The issue was related to the use of `Dispatchers.Main` for previews. The explicit use of this dispatcher causes the rendering issue in this testing framework.

**Reply by Gowsky** [reply]
[Vico v2.1.0-alpha.5](https://github.com/patrykandpatrick/vico/releases/tag/v2.1.0-alpha.5) fixes the screenshot testing issue.

**Comment by Gowsky** [comment]
[Vico v2.1.0](https://github.com/patrykandpatrick/vico/releases/tag/v2.1.0) is now available.
